### PR TITLE
feat: Implement and improve load-related intrinsics and tests

### PR DESCRIPTION
**Pull request by marktwtn**

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -4553,13 +4644,78 @@ FORCE_INLINE __m128i _mm_load_si128(const __m128i *p)
...
+// elements of dst.
+//
+//   dst[63:0] := MEM[mem_addr+63:mem_addr]
+//   dst[127:64] := MEM[mem_addr+63:mem_addr]
+//
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_load1_pd
+FORCE_INLINE __m128d _mm_load1_pd(const double *p)
+{
+#if defined(__aarch64__)
+    return vreinterpretq_m128d_f64(vdupq_n_f64(*p));
```
Comment (jserv) [inline sse2neon.h:4657]: How about `vreinterpretq_m128d_f64(vld1q_dup_f64(p))`?
[vld1q_dup_f64](https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics?search=vld1q_dup_f64) is convenient.

File: sse2neon.h [hunk 2]
```diff
@@ -876,6 +929,44 @@ FORCE_INLINE __m128i _mm_loadl_epi64(__m128i const *p)
...
+// in reverse order. mem_addr must be aligned on a 16-byte boundary or a
+// general-protection exception may be generated.
+//
+//   dst[63:0] := MEM[mem_addr+127:mem_addr+64]
+//   dst[127:64] := MEM[mem_addr+63:mem_addr]
+//
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_loadr_pd
+FORCE_INLINE __m128d _mm_loadr_pd(const double *p)
+{
+#if defined(__aarch64__)
```
Comment (jserv) [inline sse2neon.h:962]: How about the implementation below?
```c
float64x2_t v = vld1q_f64(mem_addr);
return vreinterpretq_m128d_f64(vextq_f64(v, v, 1));
```

Then, instruction `vld1q_s64` and `vextq_s64` can be applied to A32.

File: sse2neon.h [hunk 3]
```diff
@@ -787,6 +797,22 @@ FORCE_INLINE __m128 _mm_loadl_pi(__m128 a, __m64 const *p)
...
+//
+//   dst[31:0] := MEM[mem_addr+127:mem_addr+96]
+//   dst[63:32] := MEM[mem_addr+95:mem_addr+64]
+//   dst[95:64] := MEM[mem_addr+63:mem_addr+32]
+//   dst[127:96] := MEM[mem_addr+31:mem_addr]
+//
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_loadr_ps
+FORCE_INLINE __m128 _mm_loadr_ps(const float *p)
+{
+    float ALIGN_STRUCT(16) data[4] = {p[3], p[2], p[1], p[0]};
```
Comment (jserv) [inline sse2neon.h:812]: We can rewrite as following:
```c
v = vreinterpretq_f32_m128(_mm_load_ps(p));
v = vrev64q_f32(v);
return vreinterpretq_m128_f32(vextq_f32(v, v, 2));

```

File: sse2neon.h (hunk 1 above)
Comment (marktwtn) [reply sse2neon.h:4657]: That's better.

File: sse2neon.h (hunk 3 above)
Comment (marktwtn) [reply sse2neon.h]: Done.

File: sse2neon.h (hunk 1 above)
Comment (marktwtn) [reply sse2neon.h]: Done.

File: sse2neon.h (hunk 2 above)
Comment (marktwtn) [reply sse2neon.h:962]: Done.

**Pull request merged**